js/app/directives/otp.js

```diff
@@ -91,8 +91,8 @@
 							var hmacObj = new jsSHA(time, 'HEX');
 							var hmac = hmacObj.getHMAC(key, 'HEX', 'SHA-1', "HEX");
 							var offset = hex2dec(hmac.substring(hmac.length - 1));
-							var otp = (hex2dec(hmac.substr(offset * 2, 8)) & hex2dec('7fffffff')) + '';
-							otp = (otp).substr(otp.length - 6, 6);
+							var otp = (hex2dec(hmac.slice(offset * 2, offset * 2 + 8)) & hex2dec('7fffffff')) + '';
```

Comment:
It could be a good idea to move the `offset * 2` into a helper temporary variable to avoid recalculating it, but it's not a big deal on this instance.